macros/upload_individual_datasets/upload_tests.sql

```diff
@@ -11,30 +11,53 @@
-            {{ adapter.dispatch('parse_json', 'dbt_artifacts')(adapter.dispatch('column_identifier', 'dbt_artifacts')(9)) }}
-        from values
+            {{ adapter.dispatch('column_identifier', 'dbt_artifacts')(8) }},
+            {{ adapter.dispatch('parse_json', 'dbt_artifacts')(adapter.dispatch('column_identifier', 'dbt_artifacts')(9)) }},
+            {{ adapter.dispatch('column_identifier', 'dbt_artifacts')(10) }},
+            {{ adapter.dispatch('column_identifier', 'dbt_artifacts')(11) }},
+            {{ adapter.dispatch('parse_json', 'dbt_artifacts')(adapter.dispatch('column_identifier', 'dbt_artifacts')(12)) }},
+            {{ adapter.dispatch('column_identifier', 'dbt_artifacts')(13) }}
+        from ( values
         {% for test in tests -%}
+            {%- set test_name = '' -%}
+            {%- set test_type =  '' -%}
+            {%- set column_name = '' -%}
+
+            {%- if test.test_metadata is defined -%}
+                {%- set test_name = test.test_metadata.name -%}
+                {%- set test_type = 'generic' -%}
+                
+                {%- if test_name == 'relationships' -%}
+                    {%- set column_name = test.test_metadata.kwargs.field ~ ',' ~ test.test_metadata.kwargs.column_name -%}
+                {%- else -%}
+                    {%- set column_name = test.test_metadata.kwargs.column_name -%}
+                {%- endif -%}
+            {%- elif test.name is defined -%}
+                {%- set test_name = test.name -%}
+                {%- set test_type = 'singular' -%}
+            {%- endif %}
             (
                 '{{ invocation_id }}', {# command_invocation_id #}
                 '{{ test.unique_id }}', {# node_id #}
                 '{{ run_started_at }}', {# run_started_at #}
                 '{{ test.name }}', {# name #}
+                '{{ test_name }}',  {# test_name #}
+                '{{ test.config.severity }}', {# test_severity_config #}
+                '{{ column_name|escape }}', {# column_names #}
+                '{{ test_type }}', {# test_type #}
                 '{{ tojson(test.depends_on.nodes) }}', {# depends_on_nodes #}
                 '{{ test.package_name }}', {# package_name #}
                 '{{ test.original_file_path | replace('\\', '\\\\') }}', {# test_path #}
                 '{{ tojson(test.tags) }}', {# tags #}
-                {% if var('dbt_artifacts_exclude_all_results', false) %}
-                    null
-                {% else %}
-                    '{{ tojson(test) | replace("\\", "\\\\") | replace("'","\\'") | replace('"', '\\"') }}' {# all_fields #}
-                {% endif %}
+                '{{ test.unique_id }}'||'|'||'{{ test.name }}'||'|'||'{{ tojson(test.depends_on.nodes) }}'||'|'||'{{ test.package_name }}'||'|'||'{{ test.original_file_path | replace('\\', '\\\\') }}'||'|'||'{{ tojson(test.tags) }}' {# checksum #}
```

Comment:
Wouldn't we need to include info about the env and/or target.name here for the checksum? Otherwise there might be some odd things happening if differents envs load to the same table, no?

Reply:
Yeah that is a good call. Customer is only using this in production, but I see what you mean if that wasn't that case.

Reply:
I don't have a real need for now, but if we want it to be a drop-in replacement for dbt_artifacts with some enhancements, I don't see why we couldn't just keep the column

Reply:
add in the concept of environment aware variables. fixed.